code/datums/datum.dm

```diff
@@ -9,6 +9,15 @@
 	/// Datum level flags
 	var/datum_flags = NONE
 
+	/**
+	  * Any datum registered to receive observations from this datum is in this list
+	  *
+	  * Lazy associated list in the structure of `observation:registree/list of registrees`
+	  */
+	var/list/observations
```

Comment:
You have an uninitialised list here, I’m assuming it’s initialised elsewhere?

May be useful to include a wrapper to do this easily as you do with RaiseEvent